### ISSUE TITLE
Can't parse integers with 3.2.0

### ISSUE THREAD

**Issue by gahr** [Genivia/RE-flex#126]
Here's a stripped down version of a grammar I'm using

* sample.l
```lex
%option unicode
%option freespace

whitespace     [\x00\x09\x0A\x0C\x0D\x20]+
digit          [0-9]
integer        -? {digit}+
atom           [^][()<>{}/%\x00\x09\x0A\x0C\x0D\x20]+

%%

{whitespace}        { /* ignore white space */ }
{integer}           { std::cout << "int: " << str() << "\n"; }
{atom}              { std::cout << "atom: " << str() << "\n"; }
.                   { throw "undefined"; }

%%

int main(int argc, char **argv)
{
    std::istringstream input{ argv[1] };
    return Lexer(input).lex();
}
```

Compiled as `reflex -I -o sample.cpp sample.l && c++ -o sample sample.cpp -I/usr/local/include -L/usr/local/lib -lreflex`.

* RE-flex 3.1.0
```
$ ./sample 1234
int: 1234
```

* RE-flex 3.2.0
```
./sample 1234
atom: 1234
```

The generated cpp file is identical but for the different versions in `REFLEX_VERSION` and the initial `...generated by...` comment.

**Comment by genivia-inc** [comment]
I found the problem for this regression issue. I will release an update soon after some more testing, just to be sure.

**Comment by genivia-inc** [comment]
Fixed. The problem was a subtle optimization in the DFA construction that failed in very rare cases. Thanks for reporting this issue.

**Comment by gahr** [comment]
Thanks, I confirm it's fixed in 112e2de2.

**Comment by gahr** [comment]
Please don't forget to tag a 3.2.1 release, thanks!
